blockchain/v2/processor.go

```diff
@@ -137,20 +122,30 @@ func (state *pcState) purgePeer(peerID p2p.ID) {
 // handle processes FSM events
 func (state *pcState) handle(event Event) (Event, error) {
 	switch event := event.(type) {
-	case *scBlockReceived:
-		if event.height <= state.height {
+	case scFinishedEv:
+		if state.synced() {
+			return noOp, pcFinished{tdState: state.context.tdState(), blocksSynced: state.blocksSynced}
+		}
+		state.draining = true
+
+	case scBlockReceived:
+		if event.block == nil {
+			return noOp, nil
+		}
+		if event.block.Height <= state.height() {
 			return pcShortBlock{}, nil
 		}
-		err := state.enqueue(event.peerID, event.block, event.height)
+		err := state.enqueue(event.peerID, event.block, event.block.Height)
 		if err != nil {
 			return pcDuplicateBlock{}, nil
```

Comment:
It might be useful to explain somewhere the guiding principle for error handling. For example in this case we have a duplicate block which could be seen as an error but it's treated as normal operation.